service/src/main/java/com/commercetools/pspadapter/payone/notification/common/AppointedNotificationProcessor.java

```diff
@@ -52,17 +54,15 @@ protected boolean canProcess(final Notification notification) {
         final String sequenceNumber = toSequenceNumber(notification.getSequencenumber());
 
         if (findMatchingTransaction(transactions, TransactionType.CHARGE, sequenceNumber).isPresent()) {
+            // TODO: https://github.com/commercetools/commercetools-payone-integration/issues/196
+            // also: never tested (either unit nor functional)
             return listBuilder.build();
         }
 
         if (sequenceNumber.equals("1")) {
-            final Optional<Transaction> transaction =
-                    findMatchingTransaction(transactions, TransactionType.CHARGE, "0");
-            if (transaction.isPresent()) {
-                listBuilder.add(ChangeTransactionInteractionId.of(sequenceNumber, transaction.get().getId()));
-            } else {
-                addPaymentUpdatesForNewChargeTransaction(notification, listBuilder);
-            }
+
+            listBuilder.add(matchingChangeInteractionOrChargeTransaction(notification, transactions, sequenceNumber));
+
```

Comment:
1) 4d6a5cc6

2) honestly - me too. It's a legacy old code. This will be re-factored in the future as a separate task. (No you see why [having very simple one liner function](https://github.com/commercetools/commercetools-payone-integration/pull/202#discussion_r163792574) has a sense!)

2) and 3) don't want to do this in this PR. We definitely need to make complex re-factoring/re-documenting/re-formatting the sources, but as a separate task/PR.

Reply:
2) honestly - me too. It's a legacy old code. This will be re-factored in the future as a separate task. (No you see why [having very simple one liner function](https://github.com/commercetools/commercetools-payone-integration/pull/202#discussion_r163792574) has a sense!)

Reply:
2) and 3) don't want to do this in this PR. We definitely need to make complex re-factoring/re-documenting/re-formatting the sources, but as a separate task/PR.